### PR TITLE
Add tests for billing sample

### PR DESCRIPTION
**Pull request by busunkim96**
For #4505. 

Use `google-cloud-billing` to make calls to the Billing API.

### CODE REVIEW

**Review by busunkim96 — COMMENTED** [review]
@ace-n Could you give `jenkins-and-travis@python-docs-samples-tests.iam.gserviceaccount.com` the same permissions as the Node account to  `cdpe-functions-billing-test`?

@parthea This is ready for a first pass review (the tests need to have permissions sorted out).

**Comment by busunkim96** [comment]
This test now runs OK for me with my own user credentials, but fails with the service account (403 permission denied). I'm having trouble figuring out which credentials are needed (or if they can be attached to a service account).

File: billing/noxfile_config.py [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+    # Only run the tests once so tests do not interfere with each other.
+    "ignored_versions": ["2.7", "3.6", "3.7", "3.8", "3.10"],
+    # Old samples are opted out of enforcing Python type hints
+    # All new samples should feature them
+    "enforce_type_hints": False,
+    # An envvar key for determining the project id to use. Change it
+    # to 'BUILD_SPECIFIC_GCLOUD_PROJECT' if you want to opt in using a
+    # build specific Cloud project. You can also use your own string
+    # to use your own Cloud project.
+    "gcloud_project_env": "FUNCTIONS_BILLING_PROJECT",
```
Comment (parthea) [inline billing/noxfile_config.py:35]: Can you help me locate where `FUNCTIONS_BILLING_PROJECT` is defined?  I see `BUILD_SPECIFIC_GCLOUD_PROJECT` [here](https://github.com/GoogleCloudPlatform/python-docs-samples/blob/main/.kokoro/python3.9/common.cfg#L50) so I was expecting to see `FUNCTIONS_BILLING_PROJECT` in the same location.

File: billing/test_main.py [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+    request = billing.GetProjectBillingInfoRequest(name=project_name)
+    project_billing_info = cloud_billing_client.get_project_billing_info(request)
+
+    yield PROJECT_ID
+
+    # Re-enable billing in teardown
+    request = billing.UpdateProjectBillingInfoRequest(
+        name=project_name,
+        project_billing_info=billing.ProjectBillingInfo(
+
```
Comment (parthea) [inline billing/test_main.py:43]:
```suggestion
```

File: billing/test_main.py [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+from main import stop_billing
+
+
+PROJECT_ID = google.auth.default()[1]
+
+
+@pytest.fixture
+def project() -> str:
+    # This test assumes billing is enabled on the project
```
Comment (parthea) [inline billing/test_main.py:30]: Please add a comment to clarify the reason for not enabling billing at the start of the test

**Review by parthea — APPROVED** [review]
LGTM, add minor observations

File: billing/noxfile_config.py (hunk 1 above)
Comment (busunkim96) [reply billing/noxfile_config.py:35]: I added it as a new environment variable to test-env.sh (still needs to be added to test-env.tmpl.sh). There's more explanation here: https://github.com/GoogleCloudPlatform/python-docs-samples/blob/main/AUTHORING_GUIDE.md#test-environment-setup

File: billing/test_main.py (hunk 3 above)
Comment (busunkim96) [reply billing/test_main.py:30]: Fair point, if I can figure out if a service account can actually modify the billing account, I'll add the billing account as another secret and always enable bliling at the start.

**Comment by busunkim96** [comment]
@parthea I've made significant changes, could you please re-review? 

@NimishC or @tod-hilton please also review from the Billing API side.

I originally wrote the tests to disable billing on the live project, but ended up using mocks instead. The service account has enough permissions to _disable billing_, but not to _enable billing_. I'm not sure it's possible to give the service account permissions on the billing account (nor am I sure if it's a good idea) so I switched to mocks.

https://cloud.google.com/billing/docs/how-to/modify-project

Permissions required to enable billing:

> To link a project to a Cloud Billing account, you need`resourcemanager.projects.createBillingAssignment` on the project AND `billing.resourceAssociations.create` on the Cloud Billing account.

Permissions required to disable billing:

> To unlink a project from a Cloud Billing account, you need `resourcemanager.projects.deleteBillingAssignment` on the project OR `billing.resourceAssociations.delete` on the Cloud Billing account.

**Review by tod-hilton — APPROVED** [review]

**Review by parthea — APPROVED** [review]

**Pull request merged**